gctscript/modules/gct/common.go

```diff
@@ -0,0 +1,457 @@
+package gct
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	objects "github.com/d5/tengo/v2"
+	"github.com/thrasher-corp/gocryptotrader/common/file"
+	"github.com/thrasher-corp/gocryptotrader/gctscript/modules/ta/indicators"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+var commonModule = map[string]objects.Object{
+	"writeascsv": &objects.UserFunction{Name: "writeascsv", Value: WriteAsCSV},
+}
+
+// OutputDir is the default script output directory
+var OutputDir string
+
+// WriteAsCSV takes in a slice matrix to save to file
+func WriteAsCSV(args ...objects.Object) (objects.Object, error) {
+	if len(args) == 0 {
+		return nil, errors.New("cannot write to file, no data present")
+	}
+
+	var bucket [][]string
+	var err error
+	var target string
+	for i := range args {
+		if args[i] == nil {
+			return nil, errors.New("data is nil")
+		}
+		var front bool
+		var temp [][]string
+		switch args[i].TypeName() {
+		case indicators.AverageTrueRange:
+			temp, err = convertATR(args[i])
+		case indicators.BollingerBands:
+			temp, err = convertBollingerBands(args[i])
+		case indicators.ExponentialMovingAverage:
+			temp, err = convertEMA(args[i])
+		case indicators.MovingAverageConvergenceDivergence:
+			temp, err = convertMACD(args[i])
+		case indicators.MoneyFlowIndex:
+			temp, err = convertMFI(args[i])
+		case indicators.OnBalanceVolume:
+			temp, err = convertOBV(args[i])
+		case indicators.RelativeStrengthIndex:
+			temp, err = convertRSI(args[i])
+		case indicators.SimpleMovingAverage:
+			temp, err = convertSMA(args[i])
+		case indicators.OHLCV:
+			temp, err = convertOHLCV(args[i])
+			front = true
+		case "string":
+			if target != "" {
+				return nil, fmt.Errorf("filename already set, extra string %v cannot be processed", args[i])
+			}
+			var ok bool
+			target, ok = objects.ToString(args[i])
+			if !ok {
+				return nil, errors.New("failed to convert incoming output to string")
+			}
+
+			if target == "" {
+				return nil, errors.New("script context details not specified")
+			}
+
+			// Removes file transversal
+			target = filepath.Base(target)
+
+			// checks to see if file is context defined, if not it will allow
+			// a client defined filename and append a date, forces the use of
+			// .csv file extension
+			switch {
+			case !strings.HasSuffix(target, ".csv") && strings.Contains(target, ".gct"):
```

Comment:
you could also use filepath.Ext() here to get the flle extension as an alternative

Reply:
Bumperino, is neater and already using the package for other helper methods